File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def check(pro, sp):
    while pro[0] < 100:
        for idx, (p,s) in enumerate(zip(pro, sp)):
            pro[idx] = p + s
    return pro
    
def solution(progresses, speeds):
    answer = []
    pro = deque(progresses)
    sp = deque(speeds)
    
    while pro:
        returned = check(pro, sp)
        co = returned.copy()
        temp = 0
        
        for i in co:
            if i >= 100:
                pro.popleft()
                sp.popleft()
                temp += 1
            else:
                break
        if temp > 0:
            answer.append(temp)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    while (progresses.length > 0){
        let count = 0;
        while (progresses[0] >= 100) {
            progresses.shift();
            speeds.shift();
            count += 1;
        }
        if (count > 0) {
            answer.push(count);
        }
        for (let i = 0; i < progresses.length; i++){
            progresses[i] += speeds[i];
        }
    }
    return answer;
}